src/Modal.js

```diff
@@ -0,0 +1,102 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+import ReactModal from 'react-modal';
+
+let rootSelector;
+
+export const setAppElement = (selector) => {
+    rootSelector = selector;
+    ReactModal.setAppElement(selector);
+};
+
+export class Modal extends Component {
+    prevStyleValues = {};
+
+    componentDidMount() {
+        const { isOpen } = this.props;
+
+        this.rootElement = document.querySelector(rootSelector);
+
+        if (isOpen) {
+            const newStyleValues = this.computeElementValues();
+
+            this.rootElement.style.top = newStyleValues.top;
+            this.rootElement.style.left = newStyleValues.left;
+            this.rootElement.style.right = newStyleValues.right;
+            this.rootElement.style.position = newStyleValues.position;
+        }
+    }
+
+    getSnapshotBeforeUpdate(prevProps) {
+        const { isOpen: wasOpen } = prevProps;
+        const { isOpen } = this.props;
+
+        if (!wasOpen && isOpen) {
+            return this.computeElementValues();
+        }
+
+        if (wasOpen && !isOpen) {
+            return {
+                top: this.prevStyleValues.top,
+                left: this.prevStyleValues.left,
+                right: this.prevStyleValues.right,
+                bottom: this.prevStyleValues.bottom,
+                position: this.prevStyleValues.position,
+            };
+        }
+
+        return null;
+    }
+
+    componentDidUpdate(prevProps, prevState, snapshot) {
+        const { isOpen: wasOpen } = prevProps;
+        const { isOpen } = this.props;
+
+        if (wasOpen !== isOpen) {
+            this.rootElement.style.top = snapshot.top;
+            this.rootElement.style.left = snapshot.left;
+            this.rootElement.style.right = snapshot.right;
+            this.rootElement.style.position = snapshot.position;
+        }
+
+        if (wasOpen && !isOpen && typeof this.previousScrollValue !== 'undefined') {
```

Comment:
I would initialize `previousScrollValue` for the same reasons.

Reply:
Done :+1: thanks!